src/rpdk/core/generate_stack_templates.py

```diff
@@ -0,0 +1,234 @@
+            )
+
+    def _get_sorted_contract_test_files(self) -> list:
+        contract_test_folder = Path(self.target_contract_test_folder_path)
+        contract_test_files = [
+            file
+            for file in contract_test_folder.glob(CONTRACT_TEST_INPUT_PREFIX)
+            if file.is_file() and file.name in self.contract_test_file_names
+        ]
+        return sorted(contract_test_files)
+
+    def clean_and_create_stack_template_folder(
+        self, stack_template: Path, stack_template_folder: Path
+    ) -> None:
+        """
+        Clean and create the stack_template folder.
+
+        This method removes the existing stack_template root folder and creates a new stack_template folder.
+
+        Args:
+            stack_template (Path): The path to the stack_template root folder.
+            stack_template_folder (Path): The path to the stack_template folder.
+        """
+        # shutil.rmtree(stack_template, ignore_errors=True)
+        stack_template_folder.mkdir(parents=True, exist_ok=True)
+
+    def create_stack_template_bootstrap(
+        self, file_location: Path, stack_template: Path
+    ) -> None:
+        """
+        Copy the contract test dependencies to the stack_template root folder.
+
+        This method copies the contract test dependency file to the stack_template root folder
+        as the stack_template bootstrap file.
+
+        Args:
+            file_location (Path): The path to the contract test folder.
+            stack_template (Path): The path to the stack_template root folder.
+        """
+        dependencies_file = file_location / CONTRACT_TEST_DEPENDENCY_FILE_NAME
+        bootstrap_file = stack_template / self.stack_template_dependency_file_name
+        if dependencies_file.exists():
+            shutil.copy(str(dependencies_file), str(bootstrap_file))
+
+    def create_stack_template_file(
+        self,
+        resource_type: str,
+        ct_file: Path,
+        stack_template_folder: Path,
+        stack_template_file_name_prefix: str,
+        count: int,
+    ) -> None:
+        """
+        Create a stack_template file based on the contract test input file.
+
+        This method generates a stack_template file in YAML format based on the provided contract test input file.
+        The stack_template file contains the resource configuration with dynamic values replaced.
+
+        Args:
+            resource_type (str): The type of the resource being tested.
+            ct_file (Path): The path to the contract test input file.
+            stack_template_folder (Path): The path to the stack_template folder.
+            stack_template_file_name_prefix (str): The prefix for the stack_template file name.
+            count (int): The count of the stack_template file being generated.
+        """
+        with ct_file.open("r") as f:
+            json_data = json.load(f)
+        resource_name = resource_type.split("::")[2]
```

Comment:
can we get a bit more careful with potential runtime errors? resource type could be configured incorrectly

Reply:
project.type_name is already using similar logic. Type name should have this pattern.